### DISCUSSION TITLE
LineChart with areaChart and curved doesn't fill data4 area

### DISCUSSION THREAD

**Discussion by DanDoW** [Abhinandan-Kushwaha/react-native-gifted-charts#402] [Q&A] [answered]
Hello!!
First of all, thanks for the great work!

I'm having an issue building a various data sets LineChart, I paint them with diferent fill colors and use the line color transparent, but the 4th data set is not filling up, is this an error anyone has?
I tried to do the code in coding playgrounds but I wasn't able....

Thanks!

**Comment by Abhinandan-Kushwaha** [comment]
Hi @DanDoW 
Can you please send the datasets and the screenshot of the chart?

**Reply by DanDoW** [reply]
![Captura de pantalla 2023-10-09 a las 9 59 20](https://github.com/Abhinandan-Kushwaha/react-native-gifted-charts/assets/55355928/6783c765-72c4-4c88-b892-c8c46f0850ad)

```
const colors = ['#3DCD88', '#FF6BC0', '#2D9CDB', '#F9C74F', '#000000'];
const lineData1 = [{value: 0},{value: 10},{value: 8},{value: 58},{value: 56},{value: 78},{value: 74},{value: 98}];
const lineData2 = [{value: 0},{value: 20},{value: 18},{value: 40},{value: 36},{value: 60},{value: 54},{value: 85}];
const lineData3 = [{value: 10},{value: 8},{value: 58},{value: 56},{value: 78},{value: 74},{value: 0},{value: 98}];
const lineData4 = [{value: 18},{value: 40},{value: 0},{value: 20},{value: 36},{value: 60},{value: 54},{value: 85}];
const lineData5 = [{value: 8},{value: 28},{value: 56},{value: 0},{value: 10},{value: 78},{value: 74},{value: 98}];
```

not the original datasets as I made a function to make them random, but it replicates the error, I also add the props for the linechart:

```
<LineChart
  areaChart
  curved
  data={lineData1}
  data2={lineData2}
  data3={lineData3}
  data4={lineData4}
  data5={lineData5}
  color1='transparent'
  color2='transparent'
  color3='transparent'
  color4='transparent'
  color5='transparent'
  startFillColor1={colors[0]}
  endFillColor1={colors[0]}
  startFillColor2={colors[1]}
  endFillColor2={colors[1]}
  startFillColor3={colors[2]}
  endFillColor3={colors[2]}
  startFillColor4={colors[3]}
  endFillColor4={colors[3]}
  startFillColor5={colors[4]}
  endFillColor5={colors[4]}
  startOpacity={0.2}
  endOpacity={0.2}
  height={350}
  spacing={35}
  maxValue={100}
  initialSpacing={20}
  // hideDataPoints
  dataPointsColor1={colors[0]}
  dataPointsColor2={colors[1]}
  dataPointsColor3={colors[2]}
  dataPointsColor4={colors[3]}
  dataPointsColor5={colors[4]}
  hideRules
  yAxisColor='transparent'
  xAxisColor='transparent'
  overflowBottom={0}
  scrollToEnd={true}
/>
```

**Comment by Abhinandan-Kushwaha** [comment, marked as answer]
Hi @DanDoW 
This is actually a bug, and I have found the fix. Will release the new version with a fix in 1 or 2 days.

**Reply by DanDoW** [reply]
Love you!
I tried to find the bug, but I wasn't able 😅
With time and experience, I'll be able to come directly with the solution 💪😁

**Comment by Abhinandan-Kushwaha** [comment]
More power to you!
BTW, the bug is in line 1148 in the file https://github.com/Abhinandan-Kushwaha/react-native-gifted-charts/blob/master/src/LineChart/index.tsx
There should be xx4 instead of xx in line 1148
![Screenshot_2023-10-11-16-24-38-68_40deb401b9ffe8e1df2f1cc5ba480b12](https://github.com/Abhinandan-Kushwaha/react-native-gifted-charts/assets/20596944/947674e2-1d89-4688-a0da-61623bccf92c)

**Reply by DanDoW** [reply]
Awesome!!
I wasn't that far, yet far enough 🤣
I knew it was on that file, and I understood some of the ramifications, but I gave up... it felt overwhelming.
I'm glad to see I was on the right track though 😁

PS: Thank you for taking the time to explain me, it helps me grow 🤗

**Comment by Abhinandan-Kushwaha** [comment]
Hi @DanDoW 👋
This has been fixed in version `1.3.11` 🎉
